packages/editor/src/lib/exports/exportToSvg.tsx

```diff
@@ -0,0 +1,128 @@
+import { TLShapeId } from '@tldraw/tlschema'
+import { assert } from '@tldraw/utils'
+import { flushSync } from 'react-dom'
+import { createRoot } from 'react-dom/client'
+import { Editor } from '../editor/Editor'
+import { TLImageExportOptions } from '../editor/types/misc-types'
+import { StyleEmbedder } from './StyleEmbedder'
+import { embedMedia } from './embedMedia'
+import { getSvgJsx } from './getSvgJsx'
+
+export async function exportToSvg(
+	editor: Editor,
+	shapeIds: TLShapeId[],
+	opts: TLImageExportOptions = {}
+) {
+	// when rendering to SVG, we start by creating a JSX representation of the SVG that we can
+	// render with react. Hopefully elements will have a `toSvg` method that renders them to SVG,
+	// but if they don't we'll render their normal HTML content into an svg <foreignObject> element.
+	const result = getSvgJsx(editor, shapeIds, opts)
+	if (!result) return undefined
+
+	// we need to render that SVG into a real DOM element that's actually laid out in the document.
+	// without this CSS and layout aren't computed correctly, which we need to make sure any
+	// <foreignObject> elements have their styles and content inlined correctly.
+	const container = editor.getContainer()
+	const renderTarget = document.createElement('div')
+	renderTarget.className = 'tldraw-svg-export'
+	renderTarget.inert = true
+	renderTarget.tabIndex = -1
+	Object.assign(renderTarget.style, {
+		position: 'absolute',
+		top: '0px',
+		left: '0px',
+		width: result.width + 'px',
+		height: result.height + 'px',
+		pointerEvents: 'none',
+		opacity: 0,
+	})
+	container.appendChild(renderTarget)
+
+	// create a react root...
+	const root = createRoot(renderTarget)
+	try {
+		// ...and render the SVG into it.
+		flushSync(() => {
+			root.render(result.jsx)
+		})
+
+		// Some operation take a while - for example, waiting for an asset to load in. We give shape
+		// authors a way to delay snap-shotting the export until they're ready.
+		await result.exportDelay.resolve()
+
+		// Extract the rendered SVG element from the react root
+		const svg = renderTarget.firstElementChild
+		assert(svg instanceof SVGSVGElement, 'Expected an SVG element')
+
+		// And apply any changes to <foreignObject> elements that we need to make. Whilst we're in
+		// the document, these elements work exactly as we'd expect from other dom elements - they
+		// can load external resources, and any stylesheets in the document apply to them as we
+		// would expect them to. But when we pull the SVG into its own file or draw it to a canvas
+		// though, it has to be completely self-contained. We embed any external resources, and
+		// apply any styles directly to the elements themselves.
+		await applyChangesToForeignObjects(svg)
+
+		return { svg, width: result.width, height: result.height }
+	} finally {
+		// eslint-disable-next-line no-restricted-globals
```

Comment:
i'm gonna be a stickler here and say you should use editor.timers :)

Reply:
if i use editor.timers and the editor disposes, then these will never unmount

Reply:
ahh good point. then, maybe worth changing to be `container?.` since container might have been disposed potentially

Reply:
container wouldn't be set to `null` or anything as a result of that, right? we'd still have a reference to the node, the node just wouldn't be in the DOM any more. I don't think the `?` would do anything?

Reply:
ah yes! nvm